### DISCUSSION TITLE
为什么workflow不考虑造一个内存池？

### DISCUSSION THREAD

**Discussion by LunarStore** [sogou/workflow#1630] [Q&A]
为什么workflow不考虑造一个内存池？直接使用malloc和free会不会太暴力了，并且只有个别地方用到了引用计数的手法，大多数地方一块内存在哪里释放的也不好跟踪。源码阅读起来，回调绕来绕去的内存的生命周期也不好把握，很难受。

**Comment by Barenboim** [comment]
你好。workflow的内存分配释放都是跨线程的，直接调用malloc/free是最高效的方案。自己实现线程安全的内存池基本无法优于标准库或者jemalloc等内存分配库。

Discussions有新的问题好像没有提示，之后可以同步一个issue。
